analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd

```diff
@@ -0,0 +1,533 @@
+                )
+}
+```
+
+# Filter Data
+
+## Filter metadata
+
+```{r}
+# Define regions of the brain (using Anatomy of the Brain figure found at
+# https://www.ncbi.nlm.nih.gov/books/NBK65903/figure/CDR0000574573__205/)
+supratentorial <-
+  c(
+    "Skull",
+    "Thalamus",
+    "Temporal Lobe",
+    "Frontal Lobe",
+    "Parietal Lobe",
+    "Cerebrum",
+    "Basal Ganglia",
+    "Cranial Nerves NOS",
+    "Basal Ganglia;Temporal Lobe",
+    "Frontal Lobe;Parietal Lobe;Temporal Lobe",
+    "Parietal Lobe;Temporal Lobe",
+    "Frontal Lobe;Parietal Lobe"
+  )
+
+infratentorial <-
+  c(
+    "Cerebellum/Posterior Fossa",
+    "Brain Stem- Pons;Cerebellum/Posterior Fossa",
+    "Cerebellum/Posterior Fossa;Other locations NOS",
+    "Brain Stem",
+    "Brain Stem- Midbrain/Tectum;Ventricles",
+    "Cerebellum/Posterior Fossa;Ventricles",
+    "Cerebellum/Posterior Fossa;Spinal Cord- Cervical;Spinal Cord- Lumbar/Thecal Sac;Spinal Cord- Thoracic",
+    "Other locations NOS;Spinal Cord- Lumbar/Thecal Sac;Spinal Cord- Thoracic;Ventricles"
+  )
+
+# Filter metadata for `ATRT` and define `location_summary` based on values in
+# `primary_site`
+atrt_df <- metadata %>%
+  dplyr::filter(short_histology == "ATRT",
+                experimental_strategy == "RNA-Seq") %>%
+  dplyr::mutate(
+    location_summary = dplyr::case_when(
+      primary_site %in% infratentorial ~ "infratentorial",
+      primary_site %in% supratentorial ~ "supratentorial",
+      TRUE ~ "NA"
+    )
+  ) %>%
+  dplyr::group_by(sample_id) %>%
+  dplyr::summarize(Kids_First_Biospecimen_ID = paste(sort(unique(
+    Kids_First_Biospecimen_ID
+  )),
+  collapse = ", "),
+  Kids_First_Participant_ID,
+  location_summary,
+  age_at_diagnosis_days,
+  germline_sex_estimate, 
+  primary_site) 
+
+# Display `atrt_df`
+atrt_df %>%
+  head(n = 15)
+```
+
+## Prepare files for plotting of initial heatmap
+
+```{r}
+# Save Biospecimen Ids in a vector
+column_names <- colnames(stranded_expression)
+
+# Filter for Ids relevant to `ATRT`
+column_names <- column_names %>%
+  as.data.frame() %>%
+  dplyr::filter(. %in% atrt_df$Kids_First_Biospecimen_ID)
+
+# Make a data.frame with only the expression values for ATRT samples
+stranded_expression_df <- stranded_expression %>%
+  as.data.frame() %>%
+  dplyr::select(column_names$.)
+
+# Create a correlation matrix of the expression data
+initial_mat <- cor(stranded_expression_df, method = "pearson")
+
+# Write initial matrix to file
+readr::write_rds(initial_mat,
+                file.path(results_dir, "initial_heatmap_matrix.RDS"))
+
+# Create an annotation data.frame for the relevant annotation data
+initial_annotation_df <- atrt_df %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID") %>%
+  dplyr::select(-c(
+    sample_id,
+    Kids_First_Participant_ID,
+    primary_site,
+    age_at_diagnosis_days
+  ))
+
+# Make the annotation data.frame a Heatmap Annotation object
+initial_ha_atrt <- HeatmapAnnotation(df = initial_annotation_df)
```

Comment:
I would probably save the `data.frame` instead of the output of `ComplexHeatmap::HeatmapAnnotation` because it's more flexible downstream and you don't need to use the `ComplexHeatmap` package here.